### ISSUE TITLE
webpack ts-loader error

### ISSUE THREAD

**Issue by fpsqdb** [Borewit/music-metadata#129]
**Bug description**
Shows following error when compile with webpack with ts-loader,
```
WARNING in ../node_modules/music-metadata/lib/ParserFactory.d.ts
Module build failed (from ../node_modules/ts-loader/index.js):
Error: Typescript emitted no output for D:\egret resdepot\resdepot-electron\node_modules\music-metadata\lib\ParserFactory.d.ts. By default, ts-loader will not compile .ts files in node_modules.
You should not need to recompile .ts files there, but if you really want to, use the allowTsInNodeModules option.
See: https://github.com/Microsoft/TypeScript/issues/12358
    at successLoader (D:\egret resdepot\resdepot-electron\node_modules\ts-loader\dist\index.js:41:15)
    at Object.loader (D:\egret resdepot\resdepot-electron\node_modules\ts-loader\dist\index.js:21:12)
 @ ../node_modules/music-metadata/lib sync ^\.\/.*$ ./ParserFactory.d.ts
 @ ../node_modules/music-metadata/lib/ParserFactory.js
 @ ../node_modules/music-metadata/lib/index.js

WARNING in ../node_modules/music-metadata/lib/aiff/AiffParser.d.ts
Module build failed (from ../node_modules/ts-loader/index.js):
Error: Typescript emitted no output for D:\egret resdepot\resdepot-electron\node_modules\music-metadata\lib\aiff\AiffParser.d.ts. By default, ts-loader will not compile .ts files in node_modules.
You should not need to recompile .ts files there, but if you really want to, use the allowTsInNodeModules option.
See: https://github.com/Microsoft/TypeScript/issues/12358
    at successLoader (D:\egret resdepot\resdepot-electron\node_modules\ts-loader\dist\index.js:41:15)
    at Object.loader (D:\egret resdepot\resdepot-electron\node_modules\ts-loader\dist\index.js:21:12)
 @ ../node_modules/music-metadata/lib sync ^\.\/.*$ ./aiff/AiffParser.d.ts
 @ ../node_modules/music-metadata/lib/ParserFactory.js

....and more
```

**Environment**
* typescript: 3.0.1
* ts-loader: 4.4.2
* webpack: 4.16.5
* webpack-cli: 3.1.0

This is a blocking issue for #107.

**Comment by Borewit** [comment]
I don't understand why those are ending up in WebPack in the first place. As WebPack is stating, there is indeed no recompile anything in the dependencies (node_modules).

But I run into the same issue.

I solved this issue by adding the [ignore-loader](https://github.com/Borewit/webamp/blob/1070c15807c1296a26bc8aa2d0e8392ee6f883e2/package.json#L80)

Use it to ignore the `d.ts` TypeScript definition files.

```JavaScript
      {
        test: /\.d\.ts$/,
        use: [
          {
            loader: "ignore-loader"
          }
        ]
      }
```

As done [here](https://github.com/Borewit/webamp/blob/1070c15807c1296a26bc8aa2d0e8392ee6f883e2/config/webpack.common.js#L39-L46).

**Comment by fpsqdb** [comment]
@Borewit Thanks, it works after add `.d.ts ` to exclude for `ts-loader`.
```
            {
                test: /\.d\.ts$/,
                loader: "ignore-loader"
            },
            {
                test: /\.ts(x?)$/,
                exclude: [/\.d\.ts$/],
                loader: 'ts-loader'
            }
```
The strange thing is it only happen when use `music-metadata`,  other typescript lib (I use `antd` in my project) don't have this issue.

**Comment by Borewit** [comment]
Hopefully fixed in [v2.6.0](https://github.com/Borewit/music-metadata/releases/tag/v2.6.0)
